libr/include/r_util/r_str.h

```diff
@@ -19,6 +19,51 @@ typedef enum {
 	R_STRING_ENC_GUESS = 'g',
 } RStrEnc;
 
+// r_strf API can be used for short strings that are needed for a short period
+// of time. They are allocated on the stack as needed, by making sure the total
+// size of allocated strings does not become too big. If that is possible (e.g.
+// the input is not bounded and/or comes from the user) you should use the
+// malloc as usual.
+//
+// Use r_strf when you just need one string at a time. NOTE: Every time you use
+// r_strf, the previously created string can be replaced with the new one, so
+// make sure to use it only when you are done with the previous string
+//
+// Use r_strf_cont when you need to use multiple strings at a time (e.g.
+// multiple strings passed to the same function). NOTE: each string allocated
+// with r_strf_cont allocates other space on the stack, so be aware and use it
+// only for known short strings and not when inside loops as you risk of
+// allocating too much stuff.
+struct r_strf_t {
+	char *buf;
+	char *old_buf;
+	size_t sz;
+	size_t old_sz;
+	size_t total_sz;
+	int tmp;
+};
+
+#define R_STRF_FRAME r_strf_frame__
+#define r_strf_frame() struct r_strf_t R_STRF_FRAME = { NULL, NULL, 0, 0, 0 }
+#define r_strf(fmt, ...) ( \
+	R_STRF_FRAME.tmp = snprintf (R_STRF_FRAME.buf, R_STRF_FRAME.sz, fmt, __VA_ARGS__), \
+	R_STRF_FRAME.sz = R_STRF_FRAME.tmp >= R_STRF_FRAME.sz ? R_STRF_FRAME.tmp + 32 : R_STRF_FRAME.sz, \
```

Comment:
`.tmp + 32` needs to be checked for overflow as well

Reply:
why 32

Reply:
"and that's a magic numberrr" :D

Reply:
kill it